dev-tools/ls-apis/src/system_apis.rs

```diff
@@ -826,17 +860,17 @@ impl<'a> ServerComponentsTracker<'a> {
             return;
         }
 
-        if let Some((previous, _)) = self.api_producers.insert(
-            api.client_package_name.clone(),
-            (server_pkgname.clone(), dep_path.clone()),
-        ) {
+        if let Some(prev_dep_path) = self
+            .api_producers
+            .entry(api.client_package_name.clone())
+            .or_default()
+            .insert(server_pkgname.clone(), dep_path.clone())
+        {
             self.errors.push(anyhow!(
-                "API for client {} appears to be exported by multiple \
-                 components: at least {} and {} ({:?})",
+                "API for client {} appears to be exported by {server_pkgname} \
+                 via multiple dependency paths: at least {prev_dep_path:?} and \
+                 {dep_path:?}",
```

Comment:
This does seem strange, but is it actually a problem?  This definitely happens for the way components consume a client.  There, we keep track of a `Vec<DepPath>` and we iterate over that when people use `--show-deps` on the command line.

It also seems fine to not deal with this now and fix it later if it comes up.

Reply:
Changed this to just use a `Vec` in https://github.com/oxidecomputer/omicron/pull/7843/commits/dc84a1bf0ed849a9b2f066213caf86822531d2f4. In practice, it turns out to not matter as we just never actually use the dep paths, just the list of server component names. But I assume we'll use this in the future...